server/views/pages/createGoals/index.njk

```diff
@@ -123,28 +123,29 @@
                 errorMessage: errors | findError('targetCompletionDate')
               }) }}
 
-            <div class="govuk-form-group">
-              <!-- TODO: Populate the stepNumber object to relect the correct number on the UI -->
-              {% set stepTitleLabel = 'Step ' + form.stepNumber %}
-              <!-- TODO: Populate the stepTitle object to reflect the correct step title on the UI -->
-              {{ govukCharacterCount({
-                name: "stepTitle",
-                id: "stepTitle",
-                value: form.stepTitle,
-                maxlength: 512,
-                rows: 2,
-                label: {
-                  text: stepTitleLabel,
-                  classes: "govuk-label--s",
-                  isPageHeading: false
-                },
-                hint: {
-                  text: "Describe this step"
-                },
-                errorMessage: errors | findError('stepTitle')
-              }) }}
-
-            </div>
+            {% for step in form.steps %}
+              <div class="govuk-form-group">
+                <!-- TODO: Populate the stepNumber object to relect the correct number on the UI -->
+                {% set stepTitleLabel = 'Step ' + form.stepNumber %}
+                <!-- TODO: Populate the stepTitle object to reflect the correct step title on the UI -->
+                {{ govukCharacterCount({
+                  name: "stepTitle",
+                  id: "stepTitle",
+                  value: form.goals[0].steps[0].title,
```

Comment:
As with my comment above re: the `name` and `id` fields, you'll need to do it here as well
But we need to use the zero based loop index for the `step` index
Something like this I think:
```
                  name: "goals[0].step[" + loop.index0 + "].title",
                  id: "goals[0].step[" + loop.index0 + "].title",
                  value: {{ form.goals[0].steps[loop.index0].title }},
```

Also, even though I've referred to it as `loop` here, it might be worth setting the loop variable as something like `stepLoop` or similar, because in the next set of stories when we do "add another goal", then we'll have a nested loop (ie. we'll be looping through the goals, and within that loop we'll be looping through the steps)